transmogrifier/sources/transformer.py

```diff
@@ -111,18 +111,21 @@ def _write_timdex_records_to_json_file(self, output_file: str) -> int:
             while record:
                 file.write(
                     json.dumps(
-                        asdict(record, filter=lambda attr, value: value is not None),
+                        asdict(
+                            record,
+                            filter=lambda attr, value: value is not None,  # noqa: ARG005
+                        ),
```

Comment:
Just a suggestion: could use a leading `_` underscore to indicate the `attr` variable is not needed, and then remove the linting ignore:

```python
asdict(record, filter=lambda _attr, value: value is not None),
```

Or arguably even more explicit:
```python
asdict(record, filter=lambda _, value: value is not None),
```